lightning/src/ln/peer_handler.rs

```diff
@@ -961,7 +961,7 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 								if peer.pending_read_is_header {
 									let msg_len = try_potential_handleerror!(peer,
 										peer.channel_encryptor.decrypt_length_header(&peer.pending_read_buffer[..]));
-									peer.pending_read_buffer = Vec::with_capacity(msg_len as usize + 16);
+									if peer.pending_read_buffer.capacity() > 8192 { peer.pending_read_buffer = Vec::new(); }
```

Comment:
May as well move this out one level to avoid repeating the logic.

Reply:
Ah, right. I mistakenly assumed `decrypt_length_header` was doing the read, which couldn't be the case since it doesn't take a `&mut`. 😛